stat/distuv/triangle.go

```diff
@@ -0,0 +1,192 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Triangle represents a triangle distribution (https://en.wikipedia.org/wiki/Triangular_distribution).
+type Triangle struct {
+	a, b, c float64
+	Source  *rand.Rand
+}
+
+// Triangle constructs a new triangle distribution with lower limit a, upper limit b, and mode c.
+// Constraints are a < b and a ≤ c ≤ b.
+// This distribution is uncommon in nature, but may be useful for simulation.
+func NewTriangle(a, b, c float64) Triangle {
```

Comment:
The documentation says that it is acceptable for `c` to equal `a` or `b`. However, the coding of the functions returns non-sensical answers (divide by `a-c`) for instance. One of these needs to be fixed.